File: server/nodejs/src/business/domain/payments/payment-details/operation-output.ts
import { PaymentStatusOutput } from './payment-status-output';
import { AmountOfMoney } from './amount-of-money';
import { OperationPaymentReferences } from './operation-payment-references';
import { PaymentReferences } from './payment-references';

export type OperationOutput = {
    amountOfMoney?: AmountOfMoney | null;
    id?: string | null;
    operationReferences?: OperationPaymentReferences | null;
    paymentMethod?: string | null;
    references?: PaymentReferences | null;
    status?: string | null;
    statusOutput?: PaymentStatusOutput | null;
};

File: server/ruby/app/business/domain/payments/payment_details/operation_output.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class OperationOutput
          attr_accessor :amount_of_money, :id, :operation_references, :payment_method,
                        :references, :status, :status_output

          def initialize(amount_of_money: nil, id: nil, operation_references: nil,
                         payment_method: nil, references: nil, status: nil, status_output: nil)
            @amount_of_money = amount_of_money
            @id = id
            @operation_references = operation_references
            @payment_method = payment_method
            @references = references
            @status = status
            @status_output = status_output
          end

          def to_h
            {
              'amountOfMoney' => @amount_of_money&.to_h,
              'id' => @id,
              'operation_references' => @operation_references&.to_h,
              'payment_method' => @payment_method,
              'references' => @references&.to_h,
              'status' => @status,
              'status_output' => @status_output&.to_h
            }
          end
        end
      end
    end
  end
end